### PR TITLE
feat: input message to popover and add inline

### PR DESCRIPTION
**Pull request by stefanoScalzo**
## Related Issue
Closes SAP/fundamental-styles#426

## Description
Add inline form elements with popovers. Also fixed text alignment vertically for firefox
Classes added: form-input-message-group

### Before:
Screen Shot 2019-11-28 at 3 15 55 PM
### After:
Screen Shot 2019-11-28 at 3 16 26 PM

### CODE REVIEW

**Comment by InnaAtanasova** [comment]
When you hover over the input the cursor type is pointer. Should be text

**Comment by InnaAtanasova** [comment]
Screen Shot 2019-11-29 at 2 14 12 PM

RTL messages and inputs are not aligned

File: docs/pages/components/form.md [hunk 1]
```diff
@@ -32,12 +32,11 @@ Do not use the input field if:
 - The user needs to carry out a search. In this case, use the search field.
 
 {% capture inputs %} 
-    <div class="fd-form-item">
-        <label class="fd-form-label" for="input-1a">Default Input:</label>
-        <input class="fd-input" type="text" id="input-1a" placeholder="Field placeholder text">
+    <div class="fd-form-item fd-has-margin-none fd-popover">
+        <label class="fd-form-label" for="input-1">Default input:</label>
+            <input class="fd-input fd-popover__control" type="text" id="input-1" placeholder="Field placeholder text">
```
Comment (JKMarkowski) [inline docs/pages/components/form.md]: Do we need indent there ?

File: docs/pages/components/form.md [hunk 2]
```diff
@@ -92,18 +91,22 @@ Do not use the input field if:
...
+    </div>
+    <br><br>
+    <div class="fd-form-item fd-form-item--horizontal fd-has-margin-none fd-popover">
+        <label class="fd-form-label" for="input-1">Compact input:</label>
+        <div class="fd-form-input-message-group fd-form-input-message-group--compact">
+                <input class="fd-input fd-input--compact fd-popover__control" type="text" id="input-1" placeholder="Field placeholder text" aria-label="Image label" aria-controls="popoverD2" aria-expanded="false" aria-haspopup="true">
+            <span class="fd-popover__body fd-popover__body--no-arrow fd-form-message" aria-hidden="true" id="popoverD2">Normal message</span>
+        </div>
+    </div>
+    <br /><br />
```
Comment (JKMarkowski) [inline docs/pages/components/form.md]: Can you align indents there too ?

**Comment by JKMarkowski** [comment]
Required Afrer pseudoclass looks broken
image

The border above message got different color than rest of the border
image

**Comment by fkolar** [comment]
For those inline messages we just need to make sure that inline component does not have relative possitioning, but absolute. 

We dont want when showing a message the rest of the form field to be shifted down and allow form to jump up and down.

**Comment by stefanoScalzo** [comment]
> For those inline messages we just need to make sure that inline component does not have relative possitioning, but absolute.
> 
> We dont want when showing a message the rest of the form field to be shifted down and allow form to jump up and down.
> 
> This is my current issue with styles, when you are showing inline help under the field when e.g. showing error message it will move fields under down.

they are not inline they are used with popover

**Comment by fkolar** [comment]
> > For those inline messages we just need to make sure that inline component does not have relative possitioning, but absolute.
> > We dont want when showing a message the rest of the form field to be shifted down and allow form to jump up and down.
> > This is my current issue with styles, when you are showing inline help under the field when e.g. showing error message it will move fields under down.
> 
> they are not inline they are used with popover

Good then. I though its somethign similar like `<fd-form-message>`

File: src/form-input-message-group.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+/*!
+
+.fd-form-input-message-group
+*/
+$block: #{$fd-namespace}-form-input-message-group;
+
+.#{$block} {
+  flex: 1;
+  // z-index: 2;
```
Comment (InnaAtanasova) [inline src/form-input-message-group.scss]: remove the comment

File: src/form-input-message-group.scss [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  position: relative;
+
+  // &:active {
+  //   z-index: 4;
+  // }
```
Comment (InnaAtanasova) [inline src/form-input-message-group.scss]: remove the comments

File: src/form-message.scss [hunk 5]
```diff
@@ -10,20 +10,30 @@ $block: #{$fd-namespace}-form-message;
...
+  margin-top: -0.25rem;
   font-size: var(--sapFontSmallSize);
   font-weight: normal;
   line-height: normal;
   color: var(--sapTextColor);
   overflow: hidden;
   text-overflow: ellipsis;
-  box-shadow: var(--sapContent_Shadow1);
+  border-radius: 0 !important;
+  white-space: normal !important;
```
Comment (InnaAtanasova) [inline src/form-message.scss]: The while-space is `pre-line` per specs

File: docs/pages/components/form.md [hunk 6]
```diff
@@ -32,12 +32,11 @@ Do not use the input field if:
 - The user needs to carry out a search. In this case, use the search field.
 
 {% capture inputs %} 
-    <div class="fd-form-item">
-        <label class="fd-form-label" for="input-1a">Default Input:</label>
-        <input class="fd-input" type="text" id="input-1a" placeholder="Field placeholder text">
+    <div class="fd-form-item fd-has-margin-none fd-popover">
+        <label class="fd-form-label" for="input-1">Default input:</label>
+        <input class="fd-input fd-popover__control" type="text" id="input-1" placeholder="Field placeholder text">
     </div>
```
Comment (InnaAtanasova) [inline docs/pages/components/form.md]: Why do you put `fd-popover` and `fd-popover__control` when you don't have a message?

Comment (stefanoScalzo) [reply docs/pages/components/form.md]: It was supposed to be removed.There used to be a form-message on that item. I'll remove em.

File: src/form-message.scss (hunk 5 above)
Comment (stefanoScalzo) [reply src/form-message.scss]: fixed

File: src/form-input-message-group.scss (hunk 3 above)
Comment (stefanoScalzo) [reply src/form-input-message-group.scss]: removed

File: src/form-input-message-group.scss (hunk 4 above)
Comment (stefanoScalzo) [reply src/form-input-message-group.scss]: removed

**Pull request merged**